agent/consul/state/event.go

```diff
@@ -56,11 +58,23 @@ func (e *EventPublisher) Commit() {
 				delete(e.listeners[subscription.Topic], subscription)
 			}
 		}
+
+		// Update the last published index for the topic.
+		if event.Index > e.lastIndex[event.Topic] {
+			e.lastIndex[event.Topic] = event.Index
+		}
```

Comment:
Should we panic if `event.Index` is lower/equal? That seems like a pretty huge bug if it is?

Reply:
Hm, this brings up a good point that I think there are situations where it could be equal if we get multiple events generated by one state store transaction (like with the Txn api). Maybe the solution there is to emit all the events for that topic as a batched event that contains them all with one top-level index? Getting an equal/lower index here would be really bad.

Reply:
Actually I think it's OK. I was thinking back to my original buffer which had explicit design of only one event per index but your Txn example is a great one to show why that won't work. 

So I think it's OK as it is allowing multiple to be published at each index since our eventual buffer will have to support that. I don't think we can batch inside the event since in some cases we might need to emit to two different topics etc. but the buffer will need to support a set of events for each index from possibly different topics. So this is fine, for now. We can revisit whether panic/err is appropriate when we re-do the buffer.